src/gt2/gt2encode.c

```diff
@@ -107,13 +107,24 @@ void gt2MemCopy(char* out, char const* in, int size)
 
 #else
 
+#ifdef __MINGW32__
+#define GT_ENCODE_ELEM(TYPE, b, l, args)                                                                               \
+    {                                                                                                                  \
+        if (l < sizeof(TYPE))                                                                                          \
+            return -1;                                                                                                 \
+        TYPE temp = va_arg(*args, TYPE);                                                                               \
```

Comment:
Again no need to gate this behind mingw, but it would need fixing to C89 standards, so TYPE temp; needs to go at the top of the scope. A better fix is just to add `|| defined(__MINGW32__)` next to the `|| defined(_X360)` above IMO though as it seems MINGW should take the same path as the _UNIX define does.